fred2/orienteditor.cpp

```diff
@@ -182,9 +182,27 @@ BOOL orient_editor::OnInitDialog()
 	return TRUE;
 }
 
-bool orient_editor::close(float val, const CString &str)
+bool orient_editor::close(float val, const CString &input_str)
 {
-	float diff = val - convert(str);
+	CString str;
+	str.Format("%.01f", val);
+	val = convert(str);
+
+	float input_val = convert(input_str);
+
+	float diff = val - input_val;
+	return diff > -input_threshold && diff < input_threshold;
```

Comment:
I was going to suggest changing to
```
return abs(diff) < input_threshold;
```
which IMO is easier to read, but looks like the code already uses the comparison used here.

Reply:
Eh, might as well.